jwst/stpipe/step.py

```diff
@@ -422,9 +425,16 @@ def run(self, *args):
                     'make_output_path', parent_first=True
                 )
                 for idx, result in enumerate(results_to_save):
```

Comment:
Does this this handle the case were the step returns a `ModelContainer` where each `DataModel` in the container has its `meta.filename` attribute populated?  Does it use `meta.filename` or does it try to index.

This block is currently the bug that @jemorrison is seeing when she populates `meta.filename` of each model in the container and then returns the container at the end of the step.

Reply:
For documentation purposes: This particular issue fixed in #1338